### ISSUE TITLE
Using two sets of cycles for extra prediction

### ISSUE THREAD

**Issue by yjs0704** [junyanz/pytorch-CycleGAN-and-pix2pix#650]
Great work on building such a user friendly code base. I am new to deep learning, and have learned a lot from this code base. I am now wondering that if it is reasonable to write a set of networks based on your code that can do the following: 

Given three real datasets: AA (Red apple), AB (Green apple), AC (Red banana), and build a set of networks to predict BC (Green banana) without having its real data?

I have modified the code to simultaneously train two sets of cycle-GANs (AA-AB and AA-AC), and expect the G_AA-AB(AC) = G_AA-AC(AB) = BC, but it does not really perform that way (even with an extra L1 loss using this equation, and some extra cycle-consistency losses from fake BC)..

Do you think simple modifications on top of that would solve the problem? Thanks in advance.

**Comment by junyanz** [comment]
This is an interesting idea. I am not sure if the red_apple to green_apple will work for red_banana. As 
 the color and texture statistics might be different for red_apple and red_banana.

**Comment by yjs0704** [comment]
Thanks for your reply. If an extra cycle GAN would not do the job, can you recommend me an invertible deep learning/machine learning method to "normalise" the red_banana into red_apple which is expected to turn the output green_apple back into green banana?

**Comment by junyanz** [comment]
I am not aware of it. You may need to invent some new methods.

**Comment by yjs0704** [comment]
Ah ok.. Thanks very much for your reply. I plan to try to find augmentation methods will somehow probably let the training set of apples to have some distributions of bananas first.
